### DISCUSSION TITLE
Unable to add key to Microsoft account with Edge browser and strange behaviour in Windows 10 prompt

### DISCUSSION THREAD

**Discussion by huaracheguarache** [solokeys/solo2#137] [Q&A]
During testing of my Solo 2 keys on Windows 10 with webauthn.io I noticed something strange. The Windows 10 prompt that shows up when I register my keys on the website, after I've entered my key code, keeps alternating between "Getting things ready" and "Touch your security key". Doing the same with the Solo 1 does not cause the prompt to alternate between the two messages, I just get a steady "Touch your security key". Only for a brief moment after I touch it does the text change to "Getting things ready".

Key registration succeeds in both cases, but the behaviour suggests that something is not right. This might somehow be related to the other issues people have reported on Windows 10.

**Comment by huaracheguarache** [comment]
Since my primary key died yesterday I tried adding two more Solo2 keys to my Microsoft account (I ordered a lot of keys for myself and my family). I was able to add my new primary key without any issues, but when I try adding an extra backup key (in addition to my previously added backup key) it always fails. I tried adding three new Solo2 keys, and every time the following happens:

1. I sign in through account.live.com and try to add a new key to my account.
2. I give the new key a pin code as required.
3. When I name the key and click next I get to a page with the error message: "We encountered an issue setting up your security key."

During the setup process I experience the same behaviour as in the text above where the security key prompts keep alternating between "Getting things ready" and "Touch your security key".

**Reply by huaracheguarache** [reply]
I just tested and confirmed that my old Solo Tap can be added to the account without any issues.

**Reply by msiqx9** [reply]
I received a Solo V2 a couple of weeks ago with the latest firmware installed.  I had no problem adding this key to my Microsoft account using Windows 11 Pro 22H2 with the Firefox browser using WebAuthn (FIDO2).  My Solo V1 Tap also similarly works.

**Reply by huaracheguarache** [reply]
@msiqx9 Thanks, you reminded me that I should have tested other browsers than Edge. I assumed that if I'm unable to add a security key to my Microsoft account with Microsoft's own browser then it's not even worth trying other browsers. Turns out that it actually works with Firefox.

EDIT: I've changed the title of the discussion thread to better reflect the issues I'm experiencing.

**Reply by msiqx9** [reply]
I just checked logging in to my Microsoft account with the Solo v2 key that I registered to my Microsoft account two weeks ago using Firefoz browser.  I have no problems with logging in to my Microsoft account with the Edge browser in Windows 11 Pro 22H2 using the Solo v2 key.

**Reply by huaracheguarache** [reply]
Signing in with the key works for me too with Edge, it's just adding the key to the account that fails.
